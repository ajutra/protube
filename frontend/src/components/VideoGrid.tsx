import { useState, useEffect } from 'react';
import { getEnv } from '../utils/Env';

const VideoGrid = () => {
  const [someData, setSomeData] = useState([]);

  useEffect(() => {
    fetch(getEnv().API_BASE_URL + "/someEndpoint")
      .then((res) => {
        return res.json();
      })
      .then((data) => {
        console.log(data);
        setSomeData(data);
      });
  }, []);

  return (
    <div className="row g-4">
      {someData?.map((entity) => (
        "" + entity
      ))}
    </div>
  );
};

export default VideoGrid;
