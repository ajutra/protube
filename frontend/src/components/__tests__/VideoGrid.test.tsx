import { act, render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { setupServer } from "msw/node";
import { http, HttpResponse } from "msw";

jest.mock("../../../src/utils/Env.ts");

import VideoGrid from "../VideoGrid";
import { MemoryRouter } from "react-router-dom";

const server = setupServer(
  http.get("/api/someEndpoint", () => {
    return HttpResponse.json([
      {
        id: 23,
        title: "The Cranberries - Zombie (Alt. Version)",
        author: "TheCranberriesTV",
      },
    ]);
  })
);

describe("this component", () => {
  beforeAll(() => {
    server.listen();
  });

  // reset any request handlers that are declared as a part of our tests
  // (i.e. for testing one-time error scenarios)
  afterEach(() => server.resetHandlers());
  // clean up once the tests are done
  afterAll(() => server.close());

  it("renders correctly", async () => {
    const asFragment = await act(async () => {
      return render(
        <MemoryRouter>
          <VideoGrid />
        </MemoryRouter>
      ).asFragment;
    });
    expect(asFragment()).toMatchSnapshot();
  });
});
