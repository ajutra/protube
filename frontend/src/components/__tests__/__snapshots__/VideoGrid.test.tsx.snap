// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`this component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="row g-4"
  >
    [object Object]
  </div>
</DocumentFragment>
`;
